app/importers/mahonia_mapper.rb

```diff
@@ -28,9 +28,27 @@ def fields
     end
   end
 
+  def creator_name(first_name: first, last_name: last)
+    "#{first_name} #{last_name}".strip
+  end
+
+  # return first and last names of 1-n authors
+
   def creator
-    return if metadata['author1_fname'].nil? && metadata['author1_lname'].nil?
-    Array("#{metadata['author1_fname']} #{metadata['author1_lname']}")
+    creators = []
+
+    number_of_authors = metadata.keys.select do |k|
+      k.match?(/^author[0-9]*_lname$/)
+    end.count
+
```

Comment:
I like doing this with an iterator, that I can use to grab the corresponding names, it's what first comes to mind for me as the way to create the complete names. Open to suggestions if you'd prefer something else.

Reply:
@no-reply -- any thoughts on this?

Reply:
This looks okay to me.

What happens to `author1_fname` if the export has no `author1_lname` column?

Reply:
@no-reply it didn't seem to me like a good idea to make creators with just first names, but maybe that shouldn't be my decision.

Reply:
@no-reply oh nvm, I DID go ahead and use whatever is available. If there is only the first name, we use it/only the last name, we use it.

Reply:
I'm not really clear on how BePress is handling "first name" and "last name". I don't really know how they will handle names that don't fit into this rather narrow cultural assumption.

Maybe it's not a concern? Given the size of the dataset, I sort of imagine OHSU could handle exceptions on an individual basis.

Reply:
It might be worth mentioning to Kyle, or maybe in the documentation, that this is the choice we made. Even perhaps the fact that they're formatted as they are: first, space, last.

Reply:
Yeah, let's document and I can discuss with Kyle in the time we have already planned for importer review.